powdr_cli/src/main.rs

```diff
@@ -374,23 +371,46 @@ fn write_params_to_fs(params: &[u8], output_dir: &Path) {
     log::info!("Wrote params.bin.");
 }
 
+fn compile_with_csv_export<T: FieldElement>(
+    file: String,
+    output_directory: String,
+    inputs: String,
+    force: bool,
+    prove_with: Option<Backend>,
+    export_csv: bool,
+    csv_mode: CsvRenderMode,
+) {
+    let result = compile_pil_or_asm::<T>(
+        &file,
+        split_inputs(&inputs),
+        Path::new(&output_directory),
+        force,
+        prove_with,
+    );
+
+    if export_csv {
+        // Compilation result is None if the ASM file file has been compiled before
+        if let Some(compilation_result) = result {
+            let csv_path = Path::new(&output_directory).join("columns.csv");
+            export_columns_to_csv::<T>(
+                compilation_result.constants,
+                compilation_result.witness,
+                &csv_path,
+                csv_mode,
+            );
+        }
+    }
+}
+
 fn export_columns_to_csv<T: FieldElement>(
-    file: &Path,
-    dir: &Path,
+    fixed: Vec<(String, Vec<T>)>,
+    witness: Option<Vec<(String, Vec<T>)>>,
```

Comment:
Why is the witness not required?

Reply:
`compile()` doesn't produce a witness if not all constants are defined. In this case it logs `Not writing constants.bin because not all declared constants are defined (or there are none).`. The known constants would still be exported to CSV, whereas the previous code would have failed because the `commits.bin` wouldn't exist in that case.